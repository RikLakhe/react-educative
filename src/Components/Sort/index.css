.button-container{
    margin-top: 10px;
    margin-bottom: 10px;
}

.button-active{
    text-decoration: underline;
    border: 3px solid black;
}