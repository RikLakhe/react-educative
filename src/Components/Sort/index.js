import React from 'react'
import appConfig from '../../appConfig'

import './index.css'

const Sort = props =>{
    const {
        sortByTerm,
        sortOrderTerm,
        onSortButtonClick,
    }= props;

    return(
        <div className={"button-container"}>
        {
            appConfig.SORT_OPTIONS.map(sortItem=>{
                return(
                <button 
                    key={sortItem.key} 
                    onClick={()=>onSortButtonClick(sortItem.key)}
                    className={`${sortByTerm===sortItem.key ? "button-active" : ""}` }
                >
                    {sortItem.title}
                    <span>
                        {sortByTerm===sortItem.key && "*" }
                    </span>
                    </button>)
            })
        }
        </div>
    )
}

export default Sort;