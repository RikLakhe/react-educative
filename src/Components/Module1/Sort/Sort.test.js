import React from 'react';
import renderer from 'react-test-renderer';
import Sort from './index'

describe('Sort describe block', () => {
    const propItems ={
        sortByTerm: 'title',
        sortOrderTerm: 'asc',
        onSortButtonClick: jest.fn(),
    }
    let component;

    beforeEach(()=>{
        component = renderer.create(<Sort {...propItems} />)
    })

    it('Correct Number of Buttons',()=>{
        expect(component.root.findAllByType('button').length).toEqual(4);
    })

    it('On sort Button Click',()=>{
        component.root.findAllByType('button')[0].props.onClick();

        expect(propItems.onSortButtonClick).toHaveBeenCalledTimes(1);
        expect(propItems.onSortButtonClick).toHaveBeenCalledWith('title');
    })

    test('renders snapshot',async ()=>{
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
      })
})
