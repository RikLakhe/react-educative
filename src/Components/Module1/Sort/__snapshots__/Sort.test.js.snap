// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sort describe block renders snapshot 1`] = `
<div
  className="button-container"
>
  <button
    className="button-active"
    onClick={[Function]}
  >
    Title
    <span>
      ^
    </span>
  </button>
  <button
    className=""
    onClick={[Function]}
  >
    Author
    <span />
  </button>
  <button
    className=""
    onClick={[Function]}
  >
    Points
    <span />
  </button>
  <button
    className=""
    onClick={[Function]}
  >
    Number of Comments
    <span />
  </button>
</div>
`;
